clients/src/main/java/org/apache/kafka/clients/FetchSessionHandler.java

```diff
@@ -440,4 +440,13 @@ public void handleError(Throwable t) {
         log.info("Error sending fetch request {} to node {}: {}.", nextMetadata, node, t.toString());
         nextMetadata = nextMetadata.nextCloseExisting();
     }
+
+
+    public void maybeLogThrottledFetch(FetchResponse response, short version) {
```

Comment:
I am thinking where should be appropriate place we check throttling, if[ there is already error in response](https://github.com/linkedin/kafka/blob/ac1f961f7df7d6d44be2f2543aabe97d8778923c/clients/src/main/java/org/apache/kafka/clients/FetchSessionHandler.java#L382), checking for throttling seem meaningless, maybe we can check this only if there is no error in response?

Reply:
We chatted offline. The main reason of not calling this function directly in FetchSessionHandler::handleResponse is that we need a API version to determine whether throttling is supported and it is only availble outside FetchSessionHandler.